controllers/grafanafolder_controller.go

```diff
@@ -344,6 +348,7 @@ func (r *GrafanaFolderReconciler) onFolderCreated(ctx context.Context, grafana *
 
 func (r *GrafanaFolderReconciler) UpdateStatus(ctx context.Context, cr *v1beta1.GrafanaFolder) error {
 	cr.Status.Hash = cr.Hash()
+	cr.Status.NoMatchingInstances = false
```

Comment:
Could you, please, explain why this line is added here and not inside a function that checks for presence of matching instances? (assuming `UpdateStatus` can be called from anywhere)
Btw, I see that in different controllers it is set to different values.